File: backend/src/main/java/me/performancereservation/global/api/HealthCheckController.java
package me.performancereservation.global.api;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.ResponseBody;

@Controller
public class HealthCheckController {

  @Value("${commit.hash}")
  private String commitHash;

  @ResponseBody
  @GetMapping("/health-check")
  public String healthCheck() {
    return commitHash;
  }

}

File: backend/src/main/kotlin/me/performancereservation/api/HealthCheckController.kt
package me.performancereservation.api

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.ResponseBody

@Controller
class HealthCheckController {
    @Value("\${commit.hash}")
    private val commitHash: String? = null

    @ResponseBody
    @GetMapping("/health-check")
    fun healthCheck(): String? {
        return commitHash
    }
}